### ISSUE TITLE
jenkins build classpath issue

### ISSUE THREAD

**Issue by nittka** [shillner/unleash-maven-plugin#122]
We experience the following problem when using the unleash on Jenkins. We have project A with a test dependency on project B. A uses the class loader to obtain files packaged in B's test jar and throws an exception if more than one file is found. (Note that there may indeed by an error in our implementation, which still would not completely explain the behaviour!)
A regular build and a maven release build both succeed. An unleash release build fails because the same file is found in the same test jar (released version) in different repositories (jar:file:/home/gamma/jenkins/.m2/repository/... and jar:file:/home/gamma/.m2/repository/). /home/gamma/jenkins/.m2/ is not listed in the test class path of the maven build at all.

I reduced the unleash workflow to execute only buildReleaseArtifacts and nothing else. A local build (mvn unleash:perform) succeeds, the build on Jenkins again fails with the same error.

Do you have an idea why unleash seems to behave differently (and only on Jenkins)?

**Comment by shillner** [comment]
Hi, this error is hard to understand und maybe even harder to reproduce for debugging.
I'll need to have a look into the implementation of the build so but I can imagine that the Maven settings of the unleash build are different than the settings of the regular Jenkins build.
As I'm writing this I realize that it is really hard to get into this. I guess some additional information or even a minimal test project will be needed.

**Comment by nittka** [comment]
I guessed as much, I did not expect a solution, I hoped for a pointer to the potential explanation. If possible, I would have labelled this issue simply as question. As always, there is little time to create a reproducible example, in particular because it is unclear to me as well, which are the elements involved.

I'll leave this issue open for now. Maybe, I will find the time for a more thorough investigation (rather than just implement the obvious workaround).
